### DISCUSSION TITLE
Type caster trips on `std::vector<bool>`

### DISCUSSION THREAD

**Discussion by PreyMa** [wjakob/nanobind#181] [Q&A] [answered]
Hello,
I might have missed something, but I cannot get bindings using `std::vector<bool>` to work. This might be due to the specialization `std::vector<bool>` uses, which allows it to store data as compacted bitfields instead of individual items.

The compiler complains with the following (shortened) message:
```
> nanobind\include\nanobind\stl\detail/nb_list.h(62,35): error C2440: 'initializing': cannot convert from 'std::_Vb_reference<std::_Wrap_alloc<std::allocator<T>>>' to 'std::_Vb_reference<std::_Wrap_alloc<std::allocator<T>>> &'
        with
        [
            T=fmt::v9::detail::bigint::bigit
        ]
> nanobind\include\nanobind\stl\detail/nb_list.h(62,35): message : A non-const reference may only be bound to an lvalue
> nanobind\include\nanobind\nb_func.h(17,1): message : see reference to function template instantiation 'nanobind::handle nanobind::detail::list_caster<std::vector<T,std::allocator<T>>,Type>::from_cpp<std::vector<T,std::allocator<T>>>(std::vector<T,std::allocator<T>> &&,nanobind::rv_policy,nanobind::detail::cleanup_list *)' being compiled
        with
        [
            T=bool,
            Type=bool
        ]
...more template errors from the callers
```

I am currently working around this issue by manually converting my data to a `std::vector<unsigned char>` first. However, I would love to know, whether this is an oversight, intended or if I am simply doing something wrong. Thanks in advance. :^)

**Comment by wjakob** [comment]
It's surprising to me. `std::vector<bool>` is tested on all platforms as part of the CI test suite: https://github.com/wjakob/nanobind/blob/master/tests/test_stl_bind_vector.cpp#L9

**Reply by wjakob** [reply]
Based on your error message, it seems like there is a weird interaction with the `fmt` library.

**Comment by maximiliank** [comment, marked as answer]
See #256

**Reply by wjakob** [reply]
Fixed on master.
